src/node.rs

```diff
@@ -11,6 +11,9 @@ pub enum Node<'a> {
 
     /// An element potentially with attributes and children.
     Element(ElementNode<'a>),
+
+    /// An element with a namespace and potentially with attributes and children.
+    NamespacedElement(&'a str, ElementNode<'a>),
```

Comment:
I would expect an optional namespace member in `ElementNode`, and not a new `NamedspacedElement` variant. Is there a reason you chose this design? Does it manage to pack the variants into a smaller layout or somethign?

Reply:
No - now that I understand the code better, I think you're right. I originally thought having a seperate variant would need I would need to change existing code less, but actually it's the opposite. I'll change it.